darshan-util/pydarshan/darshan/backend/cffi_backend.py

```diff
@@ -340,6 +341,7 @@ def log_get_generic_record(log, mod_name, dtype='numpy'):
     buf = ffi.new("void **")
     r = libdutil.darshan_log_get_record(log['handle'], modules[mod_name]['idx'], buf)
     if r < 1:
+        libdutil.darshan_free(buf[0])
```

Comment:
Probably not super obvious from the code, but `darshan_log_get_record` will not return with any data malloced if `r < 1` (i.e., there's no more data or an error was encountered). I think we need to drop this `free()`. I started to wonder why you didn't see double free errors for this, but realized on this code path the output pointer is set to `NULL`, so you are just freeing a `NULL` pointer which isn't technically illegal.